root/static/scripts/release-editor/bindingHandlers.js

```diff
@@ -79,54 +58,43 @@ ko.bindingHandlers.artistCreditEditor = {
         }
     },
 
-    beforeShow: function (props, state) {
-        this.initialArtistText(reduceArtistCredit(state.artistCredit));
-    },
-
-    doneCallback: function () {
-        if (!this.changeMatchingArtists.peek()) {
+    doneCallback: function (initialArtistText) {
+        const input = document.getElementById('change-matching-artists');
+        if (!input || !input.checked) {
             return;
         }
 
         const track = this.currentTarget();
-        const matchWith = this.initialArtistText.peek();
         const artistCredit = track.artistCredit.peek();
 
         _(track.medium.release.mediums())
             .invokeMap("tracks").flatten().without(track).map("artistCredit")
             .each(function (ac) {
-                if (matchWith === reduceArtistCredit(ac.peek())) {
+                if (initialArtistText === reduceArtistCredit(ac.peek())) {
                     ac(artistCredit);
                 }
             });
-
-        this.initialArtistText('');
     },
 
     update: function (element, valueAccessor) {
         const bindingHandler = ko.bindingHandlers.artistCreditEditor;
         const entity = valueAccessor();
+        // Subscribe to the artistCredit observable so that we
+        // re-render the ArtistCreditEditor when it changes.
+        entity.artistCredit();
```

Comment:
In case you're not familiar with how knockout works, this update function runs inside a [computed observable](https://knockoutjs.com/documentation/computedObservables.html). Simply calling `entity.artistCredit()` (even ignoring the return value) registers it as a dependency, so this binding handler will run again when it changes.